allow overriding of boilerplate hooks and shell flags in catalog config (#4540)
gruntwork-io/terragrunt: docs-starlight/src/content/docs/02-features/06-catalog.md

1. See the docs for a selected module: `ENTER`.
1. Use [`terragrunt scaffold`](/docs/features/scaffold/) to render a `terragrunt.hcl` for using the module: `S`.

## Scaffold Configuration

The catalog configuration supports scaffold-specific settings that control the behavior when scaffolding modules:

> - **`disable_shell`** - Controls shell functions in boilerplate templates. When `true`, disables shell commands using `{{shell "command" "args"}}` syntax (enabled by default).

Reviewer: Make sure to update this too.

- **`disable_hooks`** - Controls hooks in boilerplate templates. When `true`, disables `before` and `after` hooks in their `boilerplate.yml` configuration (enabled by default).

> **Security Note**: Both shell functions and hooks can execute arbitrary commands on your system. Consider disabling these features when using untrusted templates or for enhanced security.

Reviewer: ```suggestion
**Security Note**: Both shell functions and hooks can execute arbitrary commands on your system. Consider disabling these features for enhanced security. Do not scaffold templates you do not trust.
```

### Configuration Precedence

> Scaffold configuration follows this precedence order (highest to lowest):

Reviewer: ```suggestion
When using scaffold via the catalog integration, configuration follows this precedence order (highest to lowest):
```
